data-science-onramp/data-ingestion/setup.py

```diff
@@ -0,0 +1,203 @@
+    return random.choice([user, user.upper(), user.lower(),
+                          "sub" if user == "Subscriber" else user,
+                          "cust" if user == "Customer" else user])
+
+
+def gender(s):
+    '''Manipulates the gender string'''
+    if s is None:
+        return None
+    return random.choice([s.upper(), s.lower(),
+                         s[0].upper() if len(s) > 0 else "",
+                         s[0].lower() if len(s) > 0 else ""])
+
+
+def convert_angle(angle):
+    '''Converts long and lat to DMS notation'''
+    if angle is None:
+        return None
+    degrees = int(angle)
+    minutes = int((angle - degrees) * 60)
+    seconds = int((angle - degrees - minutes/60) * 3600)
+    new_angle = str(degrees) + "\u00B0" + \
+        str(minutes) + "'" + str(seconds) + '"'
+    return random.choices([str(angle), new_angle],
+                          weights=[0.55, 0.45])[0]
+
+
+def create_bigquery_dataset():
+    # Create BigQuery Dataset
+    client = bigquery.Client()
+    dataset_id = f'{client.project}.{DATASET_NAME}'
+    dataset = bigquery.Dataset(dataset_id)
+    dataset.location = "US"
+    dataset = client.create_dataset(dataset)
+
+
+def write_to_bigquery(df, table_name):
+    '''Write a dataframe to BigQuery'''
+    client = bigquery.Client()
+    dataset_id = f'{client.project}.{DATASET_NAME}'
+
+    # Saving the data to BigQuery
+    df.write.format('bigquery') \
+        .option('table', f"{dataset_id}.{table_name}") \
+        .option("temporaryGcsBucket", BUCKET_NAME) \
+        .save()
+
+    print(f"Table {table_name} successfully written to BigQuery")
+
+
+def print_df(df, table_name):
+    '''Print 20 rows from dataframe and a random sample'''
+    # first 100 rows for smaller tables
+    df.show()
+
+    # random sample for larger tables
+    # for small tables this will be empty
+    df.sample(True, 0.0001).show(n=500, truncate=False)
+
+    print(f"Table {table_name} printed")
+
+
+def main():
+    # Create a SparkSession under the name "setup". Viewable via the Spark UI
+    spark = SparkSession.builder.appName("setup").getOrCreate()
+
+    upload = True  # Whether to upload data to BigQuery
+
+    # Check whether or not results should be uploaded
+    if '--test' in sys.argv:
+        upload = False
+        print("Not uploading results to BigQuery")
+    else:
+        create_bigquery_dataset()
+        print("Results will be uploaded to BigQuery")
+
+    # Ingest External Datasets
+
+    for table_name, data in EXTERNAL_DATASETS.items():
+        print(f'Creating dataframe for {table_name}')
+        df = spark.createDataFrame(pd.read_csv(data["url"]),
+                                   schema=data["schema"])
+
+        if upload:
+            write_to_bigquery(df, table_name)
+        else:
+            print_df(df, table_name)
+
+    # Check if table exists
+    try:
+        df = spark.read.format('bigquery').option('table', TABLE).load()
+    except Py4JJavaError:
```

Comment:
Add a short comment explaining why we'd return if this happens (if I understand correctly, this error will happen when you do the `--dry-run` option)